### PR TITLE
Prevent clones and pushes to disabled wiki

### PR DESCRIPTION
**Pull request by zeripath**
If the repository wiki is disabled prevent pushes and clones to the repository wiki

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: routers/repo/http.go [hunk 1]
```diff
@@ -298,6 +298,19 @@ func HTTP(ctx *context.Context) {
 		}
 	}
 
+	if isWiki {
+		// Ensure the wiki is enabled before we allow access to it
+		if _, err := repo.GetUnit(models.UnitTypeWiki); err != nil {
```
Comment (guillep2k) [inline routers/repo/http.go:303]: The `else` is missing for this condition.

Comment (zeripath) [reply routers/repo/http.go:303]: I don't understand

Comment (guillep2k) [reply routers/repo/http.go:303]: Oh, I see, `repo.GetUnit` will return an error if no unit is configured. Don't mind me then.

**Review by guillep2k — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
@zeripath Please send backport to v1.11

**Pull request merged**